src/error.rs

```diff
@@ -79,6 +80,15 @@ pub enum SierraAssertError {
     Range { ranges: Box<(Range, Range)> },
 }
 
+#[derive(Error, Debug)]
+pub enum CompilerError {
+    #[error("BoundedInt value is out of range: {:?} not within [{:?}, {:?}[", value, range.0, range.1)]
+    BoundedIntOutOfRange {
+        value: Box<BigInt>,
+        range: Box<(BigInt, BigInt)>,
```

Comment:
It may be better to `Box` the three values together as one tuple.

Reply:
I worry the same, but at the same time there's some risk in having multiple fields in nested errors, as they tend to add up and end up being bigger than the `Ok` variant, which ends up adding overhead to the happy path.
Another option is adding a little struct, but it may be overkill. Leaving it as it is is also a valid option.

Reply:
Let me know what you prefer and I'll adapt

Reply:
Let's leave it as is.